### PR TITLE
fix: SeamlessM4TFeatureExtractor stride remainder

### LINKED ISSUE #31916: SeamlessM4TFeatureExtractor fails with pad_to_multiple_of not being a multiple of stride
https://github.com/huggingface/transformers/blob/e314395277d784a34ee99526f48155d4d62cff3d/src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py#L285-L290

I tried to study and understand the code for the SeamlessM4TFeatureExtractor.
The code that I referenced above does not seem to be complete.
num_frames is already the size for the second dimension, so the last two lines would not be doing anything.
My guess is, that remainder should be subtracted from num_frames to get a size that is a multiple of stride.
And because of the padding this was never a problem.
Or is my thought process wrong?

### PR DESCRIPTION
**Pull request by TechInterMezzo**
# What does this PR do?

When the padded input length of SeamlessM4TFeatureExtractor is not a multiple of stride then the feature extraction fails.
This can happen when pad_to_multiple_of is not a multiple of stride and num_frames % stride is not zero.
The solution is to subtract the remainder from num_frame and adjust the size of some tensors.

Fixes #31916

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

@ylacombe

### CODE REVIEW

File: tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py [hunk 1]
```diff
@@ -171,6 +171,59 @@ def test_call_numpy(self):
         for enc_seq_1, enc_seq_2 in zip(encoded_sequences_1, encoded_sequences_2):
             self.assertTrue(np.allclose(enc_seq_1, enc_seq_2, atol=1e-3))
 
+    def test_call_with_padded_input_not_multiple_of_stride(self):
```
Comment (ylacombe) [inline tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py:174]: Great test and nice explanations !

File: tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py [hunk 2]
```diff
@@ -171,6 +171,59 @@ def test_call_numpy(self):
...
+
+        speech_inputs = [floats_list((1, x))[0] for x in range(800, 1400, 200)]
+        np_speech_inputs = [np.asarray(speech_input) for speech_input in speech_inputs]
+
+        # Test feature size
+        input_features = feature_extractor(
+            np_speech_inputs, pad_to_multiple_of=pad_to_multiple_of, return_tensors="np"
+        ).input_features
+        self.assertTrue(input_features.ndim == 3)
+        self.assertTrue(input_features.shape[0] == 3)
```
Comment (ylacombe) [inline tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py:193]: Tiny comment, could you also test that the attention mask behaves as expected here?

**Review by ylacombe — APPROVED** [review]
Thanks @TechInterMezzo, LGTM! great test, especially for a first PR!

File: tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py (hunk 2 above)
Comment (TechInterMezzo) [reply tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py]: I added a check for the attention mask size borrowed from another test. I forgot to run ruff the first time so I made a third commit to fix that. If you expect a test for the content of the attention mask then I am not really sure how to do that.

**Comment by TechInterMezzo** [comment]
@ylacombe Do I need to do anything else?

**Review by ArthurZucker — APPROVED** [review]
Thanks for fixing!

**Pull request merged**